### PR TITLE
[7.x] Add containsAll collection method

### PR DESCRIPTION
**Pull request by patrickomeara**
A lot of the time I want to make sure a collection has certain items, I don't mind if it is in a particular order, just that they exist.

```php
$service->subscribeUsersToIssue($issue, $users);

// before
$this->assertTrue($issue->subscribers->contains($users[0]);
$this->assertTrue($issue->subscribers->contains($users[1]);
$this->assertTrue($issue->subscribers->contains($users[2]);

// after
$this->assertTrue($issue->subscribers->containsAll($users));
```

I've taken the naming from `Str::containsAll()`

### CODE REVIEW

File: src/Illuminate/Support/Collection.php [hunk 1]
```diff
@@ -183,6 +183,42 @@ public function contains($key, $operator = null, $value = null)
...
+    public function containsAll($needles)
+    {
+        foreach ($this->getArrayableItems($needles) as $needle) {
+            if (! $this->contains($needle)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
```
Comment (JosephSilber) [inline src/Illuminate/Support/Collection.php:202]: This is O(NM).

This should either use a [`Set`](https://www.php.net/manual/en/class.ds-set.php), or just compare it with a `diff`:

```php
return new static($needles)->diff($this)->isEmpty();
```

Comment (patrickomeara) [reply src/Illuminate/Support/Collection.php:202]: Nice, this takes away the need for the strict comparison as well.

Comment (patrickomeara) [reply src/Illuminate/Support/Collection.php:202]: unfortunately `array_diff()` won't work with multi dimensional arrays.

> Two elements are considered equal if and only if (string) $elem1 === (string) $elem2. In other words: when the string representation is the same. 

PHP 7.3 and lower throw an error.

**Comment by GrahamCampbell** [comment]
Please send to 8.x. Laravel is no longer accepting new features to the collections library on 7.x.

**Pull request closed without merging**